### PR TITLE
Fix secrets usage and files cleanup

### PR DESCRIPTION
**Pull request by chinmaya-n**

### CODE REVIEW

File: charts/maildev/templates/deployment.yaml [hunk 1]
```diff
@@ -82,9 +82,11 @@ spec:
             - name: MAILDEV_HIDE_EXTENSIONS
               value: {{ .Values.maildev.config.smtp.hideExtensions | quote }}
             {{- end }}
+            {{- if .Values.maildev.config.smtp.incoming.authRequired }}
             {{- if .Values.maildev.config.smtp.incoming.username }}
             - name: MAILDEV_INCOMING_USER
               value: {{ .Values.maildev.config.smtp.incoming.username | quote }}
+            {{- end }}
```
Comment (alluen) [inline charts/maildev/templates/deployment.yaml:89]: So does this mean if auth is required, but no username is provided, you still need to provide the smtp-incoming-password?

File: charts/maildev/templates/smtp/service.yaml [hunk 2]
```diff
@@ -11,8 +11,10 @@ spec:
       targetPort: smtp
       protocol: TCP
       name: smtp
-      {{ if and (eq .Values.services.smtp.type "NodePort")  .Values.services.smtp.nodePort }}
-      nodePort:  {{ .Values.services.smtp.nodePort }}
-      {{ end }}
+      {{- if (eq .Values.services.smtp.type "NodePort") }}
+      {{- if .Values.services.smtp.nodePort }}
```
Comment (alluen) [inline charts/maildev/templates/smtp/service.yaml:15]: Why reconstruct this way? Is the `and` operator not working ? Just curious.

File: charts/maildev/templates/deployment.yaml (hunk 1 above)
Comment (chinmaya-n) [reply charts/maildev/templates/deployment.yaml:89]: Yes. That is the assumption I'm making b'se that is how the original Chart was. I assumed there is probably a default username the application assumes if nothing is provided. I'm not sure about the application behavior, so, I didn't deviate much from the original chart.

File: charts/maildev/templates/smtp/service.yaml (hunk 2 above)
Comment (chinmaya-n) [reply charts/maildev/templates/smtp/service.yaml:15]: I didn't confidently understand the AND operation. [Documentation Here](https://pkg.go.dev/text/template#hdr-Functions).

> and computes the Boolean AND of its arguments, returning the first false argument it encounters, or the last argument.

**Review by alluen — APPROVED** [review]

**Pull request merged**